health/services.py

```diff
@@ -40,13 +40,22 @@ def fn(app):
         client = app.config["HTTPCLIENT"]
         registry_url = _compute_internal_endpoint(app, endpoint)
         try:
-            status_code = client.get(registry_url, verify=False, timeout=2).status_code
+            status_code = client.get(
+                registry_url,
+                verify=False,
+                timeout=app.config.get("GUNICORN_HEALTH_CHECK_TIMEOUT", 2),
+            ).status_code
             okay = status_code == 200
             message = ("Got non-200 response for worker: %s" % status_code) if not okay else None
             return (okay, message)
         except Exception as ex:
-            logger.exception("Exception when checking worker health: %s", registry_url)
-            return (False, "Exception when checking worker health: %s" % registry_url)
+            logger.exception(
+                f"Exception when checking worker health on endpoint {registry_url} as {str(ex)}"
+            )
+            return (
+                False,
+                f"Exception when checking worker health on endpoint {registry_url} as {str(ex)}",
```

Comment:
any possibility this string conversion fails?

Reply:
I'll revert this for now.

Reply:
when fixing prod issues we want to make the least amount of changes as possible

Reply:
Agreed, one change at a time

Reply:
this should probably be fine, anything subclassed from `Exception` will have a default `__str__` or `__repr__`